product/src/main/java/com/yas/product/repository/ProductRepository.java

```diff
@@ -40,22 +40,26 @@ Page<Product> getProductsWithFilter(@Param("productName") String productName,
     List<Product> findAllByIdIn(List<Long> productIds);
 
     @Query(value = "FROM Product p WHERE p.isFeatured = TRUE "
-            + "AND p.isVisibleIndividually = TRUE "
-            + "AND p.isPublished = TRUE ORDER BY p.id ASC ")
-    Page<Product> getFeaturedProduct(Pageable pageable);
+        + "AND p.isVisibleIndividually = TRUE "
+        + "AND p.isPublished = TRUE "
+        + "AND p.id IN (:productIds) "
+        + "ORDER BY p.id ASC ")
+    Page<Product> getFeaturedProductByProductIds(@Param("productIds") List<Long> productIds, Pageable pageable);
 
-    @Query(value = "SELECT p FROM Product p LEFT JOIN p.productCategories pc LEFT JOIN pc.category c "
+    @Query(value = "SELECT distinct p FROM Product p LEFT JOIN p.productCategories pc LEFT JOIN pc.category c "
             + "WHERE LOWER(p.name) LIKE %:productName% "
```

Comment:
Please move the indexed condition `id` to the top.
Do we have an index on the product name? If so, using LOWER(p.name) may prevent the index use.